src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -122,7 +166,7 @@ trait RegularSync {
           val (newBlocks, errorOpt) = processBlocks(blocks, blockParentTd)
 
           if(newBlocks.nonEmpty){
-            context.self ! BroadcastBlocks(newBlocks)
+            handshakedPeers.keys.foreach(actor => actor ! BroadcastBlocks(newBlocks))
```

Comment:
We will send same block to the same peerActor we received it from
smart part is in peerActor and it will just ignore this message